app/_how-tos/kic-plugin-mtls.md

```diff
@@ -0,0 +1,135 @@
+  - kic
+
+tools:
+  - kic
+
+works_on:
+  - on-prem
+  - konnect
+
+entities: []
+
+tldr:
+  q: How do I enforce mTLS from a client to {{ site.base_gateway }} using {{ site.kic_product_name }}?
+  a: Create a Secret containing a CA Certificate and pass the ID of the certificate to an mTLS plugin configuration
+
+prereqs:
+  enterprise: true
+  kubernetes:
+    gateway_api: true
+  entities:
+    services:
+      - echo-service
+    routes:
+      - echo
+
+cleanup:
+  inline:
+    - title: Uninstall KIC from your cluster
+      include_content: cleanup/products/kic
+      icon_url: /assets/icons/kubernetes.svg
+---
+
+## About mTLS
+
+Mutual TLS (mTLS) is a way to secure connectivity using certificates. {{ site.base_gateway }} can look for a certificate in incoming requests and reject the request if the public key presented does not match the private key stored in {{ site.base_gateway }}.
+
+## Generate a CA certificate
+
+To use the `mtls-auth` plugin you need a CA certificate. If you don't have one, generate a new certificate using `openssl`:
+
+```bash
+openssl req -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -sha256 -days 365 -nodes \
+-subj "/C=US/ST=California/L=San Francisco/O=Kong/OU=Org/CN=www.example.com"
+```
+
+## Add the certificate to {{ site.base_gateway }}
+
+CA certificates in Kong are provisioned by creating `Secret` or `ConfigMap` resource in Kubernetes.
+
+Resources holding CA certificates must have the following properties:
+- the `konghq.com/ca-cert: "true"` label applied.
+- a `cert` or `ca.crt` data property which contains a valid CA certificate in PEM format.
+- a `kubernetes.io/ingress.class` annotation whose value matches the value of the controller's `--ingress-class` argument. By default, that value is `kong`.
+- an `id` data property which contains a random UUID.
+
+Each CA certificate that you create needs a unique ID. Any random UUID should suffice here and it doesn't have a security implication. You can use [uuidgen](https://linux.die.net/man/1/uuidgen) (Linux, OS X) or [New-Guid](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/new-guid) (Windows) to generate an ID.
+
+```bash
+CERT_ID=$(uuidgen | tr "[:upper:]" "[:lower:]")
+kubectl create secret -n kong generic my-ca-cert --from-literal=id=$CERT_ID --from-file=cert=./cert.pem
+kubectl label secret -n kong my-ca-cert 'konghq.com/ca-cert=true'
+kubectl annotate secret -n kong my-ca-cert 'kubernetes.io/ingress.class=kong'
+```
+
+## Configure the mtls-auth plugin
+
+The [mtls-auth plugin](/plugins/mtls-auth/) requires a CA Certificate ID that will be used to validate the certificate in the incoming request. In this example we disable revocation checks, but you should enable checks in a production setting.
+
+{% entity_example %}
+type: plugin
+data:
+  name: mtls-auth
+  config:
+    ca_certificates:
+    - $CERT_ID
+    skip_consumer_lookup: true
+    revocation_check_mode: SKIP
+
+  service: echo
+{% endentity_example %}
```

Comment:
This didn't work for me, I don't think my enterprise license set correctly. I tried running the cleanup and starting over, but that didn't fix it.

```
Error from server: error when creating "STDIN": admission webhook "validations.kong.konghq.com" denied the request: plugin failed schema validation: schema violation (name: 'mtls-auth' is an enterprise only plugin)
```